tools/src/main/scala/scala/scalanative/build/Build.scala

```diff
@@ -136,4 +168,18 @@ object Build {
     }
   }
 
+  private def getLastModifiedTimeMillis(path: Path): Long =
+    if (Files.exists(path)) Files.getLastModifiedTime(path).toMillis()
+    else 0L
+
+  /** Get the newest file's last modified time in millis under the given path.
+   */
+  private def getNewestMtime(path: Path): Long =
+    Files
+      .walk(path, FileVisitOption.FOLLOW_LINKS)
+      .iterator()
+      .asScala
+      .map(p => Files.getLastModifiedTime(p).toMillis())
+      .max
```

Comment:
There seemed to be some deprecation warning elevated to the errors due to `-Xfatal-warnings` flag. 
However, in this case we can just fallback to Java streams and skip converting `Stream->Iterator->scala.Iterator`: 

```scala
Files
  .walk(path, FileVisitOption.FOLLOW_LINKS)
  .map(Files.getLastModifiedTime(_))
  .max(_.compareTo(_)) // : j.u.Optional[j.n.f.a.FileTime]
```
Probably we can use resulting `Optional[FileTime]` directly - when I've tested it in REPL it had a stable hashCode